colibris/views.py

```diff
@@ -1,17 +1,208 @@
+    model = Model
+    schema_class = ModelSchema
+
+    async def get(self):
+        items = self.get_query()
+        schema = self.get_schema(many=True)
+        result = schema.dump(list(items))
+
+        return web.json_response(result)
+
+
+class CreateMixin(metaclass=_GenericMixinMeta):
+    model = Model
+    schema_class = ModelSchema
+
+    async def post(self):
+        schema = self.get_schema()
+        data = await self.get_validated_data(schema)
+
+        try:
+            item = self.model.create(**data)
+        except IntegrityError as err:
+            traceback.print_exc()
+            raise api.ServerError(code='server_error', message=str(err))
+
+        result = schema.dump(item)
+
+        return web.json_response(result, status=201)
+
+
+class RetrieveMixin(metaclass=_GenericMixinMeta):
+    model = Model
+    schema_class = ModelSchema
+
+    async def get(self):
+        schema = self.get_schema()
+        instance = self.get_object()
+
+        result = schema.dump(instance)
+
+        return web.json_response(result)
+
+
+class UpdateMixin(metaclass=_GenericMixinMeta):
+    model = Model
+    schema_class = ModelSchema
+
+    async def patch(self):
+        schema = self.get_schema(partial=True)
+        data = await self.get_validated_data(schema)
+
+        instance = self.get_object()
+        instance.update_fields(data)
+
+        try:
+            instance.save(only=data.keys())
+        except IntegrityError as err:
+            traceback.print_exc()
+            raise api.ServerError(code='server_error', message=str(err))
+
+        result = schema.dump(instance)
+
+        return web.json_response(result)
+
+    async def put(self):
+        schema = self.get_schema()
+        data = await self.get_validated_data(schema)
+
+        instance = self.get_object()
+        instance.update_fields(data)
+
+        try:
+            instance.save()
+        except IntegrityError as err:
+            traceback.print_exc()
+            raise api.ServerError(code='server_error', message=str(err))
+
+        result = schema.dump(instance)
+
+        return web.json_response(result)
+
+
+class DestroyMixin:
+    async def delete(self):
+        instance = self.get_object()
+        instance.delete_instance()
+
+        return web.json_response(status=204)
+
+
+class APIView(web.View):
+    schema_class = Schema
+
+    def get_schema(self, *args, **kwargs):
+        kwargs.update({
+            'context': {'request': self.request}
+        })
+
+        schema = self.schema_class(*args, **kwargs)
+
+        return schema
+
+    async def get_validated_data(self, schema=None):
+        if schema is None:
+            schema = self.get_schema()
+
+        json_payload = await self.get_request_payload()
+
+        try:
+            data = schema.load(json_payload)
+        except ValidationError as err:
+            raise api.InvalidRequest(code='invalid_request', message=err.messages)
```

Comment:
We'd like to have a dedicated exception for schema validation errors, such as `HTTPSchemaValidationError`.